Framework/Interaction/Rest/Customer.php

```diff
@@ -108,4 +109,56 @@ public function downloadCertificate(
             $request->getData('type')
         );
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteCertificate(
+        $request,
+        $isProduction = null,
+        $scopeId = null,
+        $scopeType = \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+    )
+    {
+        /** @var \Avalara\AvaTaxClient $client */
+        $client = $this->getClient($isProduction, $scopeId, $scopeType);
+
+        try {
+            $customerId = $this->customerHelper->getCustomerCode($request->getData('customer_id'), null, $scopeId);
+
+            //unlink request requires a LinkCustomersModel which contains a string[] of all customer ids.
+            $customerModel = new \Avalara\LinkCustomersModel();
+            $customerModel->customers = [$customerId];
+
+            //Customer(s) must be unlinked from cert before it can be deleted.
+            $unlinkResult = $client->unlinkCustomersFromCertificate(
+                $this->config->getCompanyId($scopeId, $scopeType),
+                $request->getData('id'),
+                $customerModel
+            );
+
+            $this->validateResult($unlinkResult, $request);
+
+        } catch (\Exception $e) {
+            //Swallow this error. Continue to try and delete the cert.
+            //If the deletion errors, then we'll notify the user that something has gone wrong.
+        }
+
+        //make deletion request.
+        $result = $client->deleteCertificate(
+            $this->config->getCompanyId($scopeId, $scopeType),
+            $request->getData('id')
+        );
+
+        //A successful delete request results in an empty body. This means $result is an empty array.
+        //However, the validateResult method can't handle this as a valid response.
+        //This explicit check for an empty array is to fill that hole in the validation.
+        if(is_array($result) && count($result) === 0) {
+            return $result; //result is an empty array. No error was returned from request.
+        } else {
```

Comment:
No need for an `else` statement since you return on the if statement. Avoid using `else`